Add a github-deploy plugin

torbjornvatn opened the pull request:
Finally got around to making this plugin

Commit:
Add a github-deploy plugin

judoole commented on plugins/github/github-deploy.mk:
```diff
@@ -0,0 +1,20 @@
+	# Get latest, so we can diff if there are changes
+	git fetch
+	# Check if there is staged code or changes. Exit if so
+	git diff @{push} --shortstat --exit-code ||(echo "There's git diff ⤴️, exiting..."; exit $$1)
```
This looks pretty correct, but when I ran just now I got this
```
Checking diff local against remote
fatal: no upstream configured for branch 'local_dev_fixes'
There's git diff ⤴️, exiting...
```
But it went along and deployed anyways 🤔

judoole replied:
I think this would work (tested in a local project):
```
define deploy
	echo Checking diff local against remote
	# Get latest, so we can diff if there are changes
	git fetch
	# Check if there is staged code or changes. Exit if so
	if git fetch && git diff @{push} --shortstat --exit-code; then \
		gh api repos/:owner/:repo/deployments -H "Accept: application/vnd.github.ant-man-preview+json" \
	               --method POST -F ref=":branch" -F environment="$(strip $1)" -F auto_merge=${AUTO_MERGE} \
         	      -F task="$(strip $(if $2, $2, deploy))"; \
	else \
		echo "There is a git diff ⤴️, exiting..."; \
	fi;
endef
```

Does look okey @torbjornvatn ?

judoole replied:
Got eager and made a commit for testing https://github.com/unacast/brokkr/commit/914a4c3da3f28075ff1629dee854c6fbd3486488

judoole reviewed (changes requested):
Think we need to change the `||`

judoole committed:
Use shell if/else rather than || + parantheses

`||` and parantheses leads to a subshell and Make does not pick up the exit code in the subshell
